hwy/tests/combine_test.cc

```diff
@@ -284,6 +284,170 @@ HWY_NOINLINE void TestAllConcatOddEven() {
   ForAllTypes(ForShrinkableVectors<TestConcatOddEven>());
 }
 
+class TestTruncatingResizeBitCast {
+#if HWY_TARGET != HWY_SCALAR
+ private:
+  template <class DTo, class DFrom>
+  static HWY_INLINE void DoTruncResizeBitCastTest(DTo d_to, DFrom d_from) {
+    const VFromD<DFrom> v = Iota(d_from, 1);
+    const VFromD<DTo> expected = Iota(d_to, 1);
+
+    const VFromD<DTo> actual_1 = ResizeBitCast(d_to, v);
+    HWY_ASSERT_VEC_EQ(d_to, expected, actual_1);
+
+    const VFromD<DTo> actual_2 = ZeroExtendResizeBitCast(d_to, v);
+    HWY_ASSERT_VEC_EQ(d_to, expected, actual_2);
+
+    const VFromD<DTo> actual_3 = ZeroExtendResizeBitCast(d_to, d_from, v);
+    HWY_ASSERT_VEC_EQ(d_to, expected, actual_3);
+  }
+#endif  // HWY_TARGET != HWY_SCALAR
+ public:
+  template <class T, class D>
+  HWY_NOINLINE void operator()(T /*unused*/, D d) {
+#if HWY_TARGET != HWY_SCALAR
+    const Half<D> dh;
+    DoTruncResizeBitCastTest(dh, d);
+
+    const auto v_full = Iota(d, 1);
+    const VFromD<decltype(dh)> expected_full_to_half = LowerHalf(dh, v_full);
+    HWY_ASSERT_VEC_EQ(dh, expected_full_to_half, ResizeBitCast(dh, v_full));
+    HWY_ASSERT_VEC_EQ(dh, expected_full_to_half,
+                      ZeroExtendResizeBitCast(dh, v_full));
+    HWY_ASSERT_VEC_EQ(dh, expected_full_to_half,
+                      ZeroExtendResizeBitCast(dh, d, v_full));
+
+    constexpr size_t kMaxLanes = MaxLanes(d);
+#if HWY_TARGET == HWY_RVV
```

Comment:
Can/should we use this same code on other targets, with just a target-dependent max actual pow2 (3 on RVV, 0 otherwise)?